src/django_pg_migration_tools/operations.py

```diff
@@ -70,17 +71,22 @@ def database_forwards(
         to_state: migrations.state.ProjectState,
     ) -> None:
         self._ensure_not_in_transaction(schema_editor)
-        self._ensure_no_lock_timeout_set(schema_editor)
-        self._ensure_not_an_invalid_index(schema_editor)
-        model = from_state.apps.get_model(app_label, self.model_name)
-        index_sql = str(self.index.create_sql(model, schema_editor, concurrently=True))
-        # Inject the IF NOT EXISTS because Django doesn't provide a handy
-        # if_not_exists: bool parameter for us to use.
-        index_sql = index_sql.replace(
-            "CREATE INDEX CONCURRENTLY", "CREATE INDEX CONCURRENTLY IF NOT EXISTS"
-        )
-        schema_editor.execute(index_sql)
-        self._ensure_original_lock_timeout_is_reset(schema_editor)
+        if router.allow_migrate(
+            schema_editor.connection.alias, app_label, **self.hints
+        ):
```

Comment:
🐼  Given the size of the indented block below, would it be clearer to do this instead?

```python
if not router.allow_migrate([...]):
    return
```

Reply:
+1

Reply:
Fair enough -> 6f30632a998687280364a47e82ac0290587642a6